pandas/tests/frame/test_analytics.py

```diff
@@ -1966,12 +1966,35 @@ def test_n(self, df_strings, n, order):
             tm.assert_frame_equal(result, expected)
 
     def test_n_error(self, df_strings):
-        # b alone raises a TypeError
-        df = df_strings
-        with pytest.raises(TypeError):
-            df.nsmallest(1, 'b')
-        with pytest.raises(TypeError):
-            df.nlargest(1, 'b')
+        df = pd.DataFrame(
```

Comment:
create this as another fixture (e.g. *like* df_strings*, but name something different) and use parametrize